### PR TITLE
cor/codespaces-release

### PR DESCRIPTION
**Pull request by cor**
## Issue
https://app.clickup.com/t/2au272n

## Description
This should be the final PR for the first official Codespaces release for our team! Mostly contains documentation on how to use everything.

## Checklist

- [x] I have updated the `book/` to reflect changes made by this PR _(if applicable)_

### CODE REVIEW

File: book/src/codespaces/using-codespaces.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Using Codespaces
+
+*easy peasy lemon squeezy*
```
Comment (cor) [inline book/src/codespaces/using-codespaces.md:3]: Do we _really_ want to say "easy peasy lemon squeezy"?

Comment (cor) [reply book/src/codespaces/using-codespaces.md:3]: Yes, yes we do.

**Review by hussein-aitlahcen — DISMISSED** [review]

File: book/src/codespaces.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Codespaces
+
+---
+
+At Composable Finance, we use [Codespaces](https://github.com/features/codespaces) in order to provide **blazing fast cloud developer environments.** This means that we all use the same [Dockerized devcontainer environment](https://github.com/ComposableFi/composable/tree/main/.devcontainer). It is moves all of the resource-intensive processing such as code compilation and analysis to the cloud, while the UI is still rendered locally, providing a **fast, responsive, and consistent user experience**.
+
+This gives us the following advantages
```
Comment (benluelo) [inline book/src/codespaces.md:7]:
```suggestion
This gives us the following advantages:
```

File: book/src/codespaces.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+- All configuration of developer tooling is already done for you and shared with your team. If something works for one person, it works for everyone. This ensures that you can instantly start coding. No more "It works on my machine".
+- You leverage fast (`16-core CPU`, `32GB RAM`) machines to compile and run your code. Which speeds up your compilation times, and saves you resources on your main machine.
+- You don't have to worry about using the wrong versions of dependencies, build systems, or other tools, since everyone uses the same dependencies that have been defined in code, and are identical to the ones used in our CI pipelines.
+- If you come up with an improvement to our development experience, you can open an PR for it and then the entire team can take advantage of it.
```
Comment (benluelo) [inline book/src/codespaces.md:12]:
```suggestion
- If you come up with an improvement to our development experience, you can open a PR for it and then the entire team can take advantage of it.
```

File: book/src/codespaces.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Codespaces
+
+---
+
+At Composable Finance, we use [Codespaces](https://github.com/features/codespaces) in order to provide **blazing fast cloud developer environments.** This means that we all use the same [Dockerized devcontainer environment](https://github.com/ComposableFi/composable/tree/main/.devcontainer). It is moves all of the resource-intensive processing such as code compilation and analysis to the cloud, while the UI is still rendered locally, providing a **fast, responsive, and consistent user experience**.
+
+This gives us the following advantages
+
+- All configuration of developer tooling is already done for you and shared with your team. If something works for one person, it works for everyone. This ensures that you can instantly start coding. No more "It works on my machine".
+- You leverage fast (`16-core CPU`, `32GB RAM`) machines to compile and run your code. Which speeds up your compilation times, and saves you resources on your main machine.
```
Comment (benluelo) [inline book/src/codespaces.md:10]:
```suggestion
- You leverage high performance (`16-core CPU`, `32GB RAM`) machines to compile and run your code, speeding up your compilation times and saving you resources on your main machine.
```

**Review by PoisonPhang — DISMISSED** [review]

File: book/src/codespaces/substrate.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+*Substrate pallet development*
+
+---
+
+## Running tests
+
+Open the integrated terminal, and run
+
+```bash
+cargo test
```
Comment (benluelo) [inline book/src/codespaces/substrate.md:17]:
```suggestion
cargo test --workspace
```

**Review by benluelo — DISMISSED** [review]

**Review by PoisonPhang — DISMISSED** [review]

**Review by vimukthi-git — APPROVED** [review]
:rocket: 
NIt: may be the images can be moved to a separate directory from the md files?

**Review by 0xBrainjar2 — APPROVED** [review]

**Review by vimukthi-git — APPROVED** [review]

**Pull request merged**